### PR TITLE
:book: Add Metal3 to list for beta release notification

### PR DESCRIPTION
**Pull request by nawazkh**
**What this PR does / why we need it**:
- This PR add metal-3 to the list of CAPI providers in release-tasks.md.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -399,6 +399,7 @@ We should inform at least the following providers via a new issue on their respe
  - Kubemark: https://github.com/kubernetes-sigs/cluster-api-provider-kubemark/issues/new
  - Kubevirt: https://github.com/kubernetes-sigs/cluster-api-provider-kubevirt/issues/new
  - IBMCloud: https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/new
+ - Metal-3: https://github.com/metal3-io/cluster-api-provider-metal3/issues/new
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:402]:
```suggestion
 - Metal3: https://github.com/metal3-io/cluster-api-provider-metal3/issues/new
```

**Comment by killianmuldoon** [comment]
/retitle book Add <etal3 to list for beta release notification

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

**Review by killianmuldoon — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**